libcxx/include/regex

```diff
@@ -4587,28 +4587,36 @@ public:
 
   // element access:
   _LIBCPP_HIDE_FROM_ABI difference_type length(size_type __sub = 0) const {
-    _LIBCPP_ASSERT_UNCATEGORIZED(ready(), "match_results::length() called when not ready");
+    // If the match results are not ready, this will return `0`.
+    _LIBCPP_ASSERT_PEDANTIC(ready(), "match_results::length() called when not ready");
```

Comment:
Since this likely gives the wrong result and is explicitly done by the developer I would make this a stronger assert.

Reply:
if we do `pedantic` only for no-ops I would rename it. I have a different view what the word pedantic means.